spec/controllers/sections_controller_spec.rb

```diff
@@ -0,0 +1,145 @@
+          it 'redirects to educator\'s first section' do
+            make_request('garbage ids rule')
+            expect(response).to redirect_to(section_path(educator.sections[0]))
+          end
+        end
+
+        context 'params for section assigned to educator' do
+          it 'is successful' do
+            make_request(first_section.id)
+            expect(response.status).to eq 200
+          end
+          it 'assigns correct sections to drop-down' do
+            make_request(first_section.id)
+            expected_section_ids = [first_section, second_section].map(&:id)
+            sections = extract_serialized_ids(controller, :sections)
+            expect(sections).to match_array(expected_section_ids)
+          end
+
+          context 'when there are no students' do
+            it 'assigns students to empty' do
+              make_request(second_section.id)
+              student_ids = extract_serialized_ids(controller, :students)
+              expect(student_ids).to be_empty
+            end
+          end
+
+          context 'when there are students' do
+            it 'assigns rows to a non-empty array' do
+              make_request(first_section.id)
+              expected_student_ids = [first_student, second_student, third_student].map(&:id)
+              student_ids = extract_serialized_ids(controller,:students)
+              expect(student_ids).to match_array(expected_student_ids)
+            end
+          end
+        end
+
+        context 'section not accessible to educator' do
+          let(:another_section) { FactoryGirl.create(:section) }
+          it 'redirects' do
+            make_request(another_section.id)
+            expect(response.status).to eq 302
+          end
+        end
+      end
+    end
+
+    context 'admin educator logged in' do
+      let(:admin_educator) { FactoryGirl.create(:educator, :admin, school: school) }
+      
+      before { sign_in(admin_educator) }
+
+      context 'when requesting a section inside their school' do
+        it 'is successful' do
+          make_request(first_section.id)
+          expect(response.status).to eq 200
+        end
+
+        it 'assigns correct sections to drop-down' do
+          make_request(first_section.id)
+          expected_section_ids = [first_section, second_section, third_section].map(&:id)
+          sections = extract_serialized_ids(controller, :sections)
+          expect(sections).to match_array(expected_section_ids)
+        end
+      end 
+
+      context 'when requesting section outside their school' do
+        it 'redirects' do
+          make_request(other_school_section.id)
+          expect(response.status).to eq 302
+        end
+      end
+    end
+    
+    context 'districtwide educator logged in' do
+      let(:dw_educator) { FactoryGirl.create(:educator, districtwide_access: true, school: school) }
+  
+      before { sign_in(dw_educator) }
+
+      context 'when requesting a section inside their school' do
```

Comment:
District-wide educators don't need to be assigned to a school; see `Educator#has_school_unless_districtwide` under `educator.rb`. Can you adjust these context blocks to account for that?

Reply:
@alexsoble Changed it a bit to remove the school for district wide admins. Point is regardless of whether they have a school assigned, they always see all sections districtwide.